nntrainer/models/neuralnet.h

```diff
@@ -345,6 +345,18 @@ class NeuralNetwork : public ml::train::Model {
    */
   void printPreset(std::ostream &out, unsigned int preset);
 
+#ifdef PROFILE
+  /**
+   * @brief query profile result
+   *
+   * @return double profile result
+   */
+  long int profileResult() { return time_taken; };
+
+private:
+  long int time_taken;
```

Comment:
add some comment here? 
and how about forward_time_taken? as there will be a backward timing variable later too.

Reply:
Regarding this,  I encapsulated and modularized the profiler into an inner class. Please take a look. Thanks for the suggestion